x-pack/legacy/plugins/siem/public/components/ml_popover/jobs_table/job_switch.tsx

```diff
@@ -48,17 +48,18 @@ export const JobSwitchComponent = ({
 }: JobSwitchProps) => {
   const [isLoading, setIsLoading] = useState(false);
   const handleChange = useCallback(
-    e => {
+    async e => {
       setIsLoading(true);
-      onJobStateChange(job, job.latestTimestampMs || 0, e.target.checked);
+      await onJobStateChange(job, job.latestTimestampMs || 0, e.target.checked);
+      setIsLoading(false);
     },
     [job, setIsLoading, onJobStateChange]
   );
 
   return (
     <EuiFlexGroup justifyContent="spaceAround">
       <EuiFlexItem grow={false}>
-        {isSiemJobsLoading || isLoading || isJobLoading(job.jobState, job.datafeedId) ? (
+        {isSiemJobsLoading || isLoading || isJobLoading(job.jobState, job.datafeedState) ? (
```

Comment:
This seemed like a typo that'd always been there. I didn't see much of a change here, I guess the bug was that it would _not_ show as loading when in fact the data feed still was (loading).